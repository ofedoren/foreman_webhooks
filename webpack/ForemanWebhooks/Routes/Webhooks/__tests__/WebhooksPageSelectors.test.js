import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';

import {
  selectWebhooks,
  selectPage,
  selectPerPage,
  selectSearch,
  selectSort,
  selectHasData,
  selectHasError,
  selectIsLoading,
  selectSubtotal,
  selectMessage,
} from '../WebhooksPageSelectors';

import {
  stateFactory,
  webhooks,
} from '../WebhooksIndexPage/__tests__/WebhooksIndexPage.fixtures';

const state = stateFactory({
  results: webhooks,
  sort: { by: 'name', order: 'DESC' },
  page: 1,
  perPage: 1,
  search: 'name ~ foo',
  subtotal: 42,
  message: { type: 'error', text: 'This is error' },
});

const fixtures = {
  'should return webhooks': () => selectWebhooks(state),
  'should return page': () => selectPage(state),
  'should return perPage': () => selectPerPage(state),
  'should return search': () => selectSearch(state),
  'should return sort': () => selectSort(state),
  'should return hasData': () => selectHasData(state),
  'should return hasError': () => selectHasError(state),
  'should return isLoading': () => selectIsLoading(state),
  'should return subtotal': () => selectSubtotal(state),
  'should return message': () => selectMessage(state),
};

describe('WebhooksPage selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
