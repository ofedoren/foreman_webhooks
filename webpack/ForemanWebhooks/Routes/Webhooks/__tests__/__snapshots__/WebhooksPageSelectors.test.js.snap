// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebhooksPage selectors should return hasData 1`] = `false`;

exports[`WebhooksPage selectors should return hasError 1`] = `false`;

exports[`WebhooksPage selectors should return isLoading 1`] = `true`;

exports[`WebhooksPage selectors should return message 1`] = `
Object {
  "text": "This is error",
  "type": "error",
}
`;

exports[`WebhooksPage selectors should return page 1`] = `1`;

exports[`WebhooksPage selectors should return perPage 1`] = `1`;

exports[`WebhooksPage selectors should return search 1`] = `"name ~ foo"`;

exports[`WebhooksPage selectors should return sort 1`] = `
Object {
  "by": "name",
  "order": "DESC",
}
`;

exports[`WebhooksPage selectors should return subtotal 1`] = `42`;

exports[`WebhooksPage selectors should return webhooks 1`] = `
Array [
  Object {
    "canDelete": true,
    "canEdit": true,
    "enabled": true,
    "id": 1,
    "name": "my-webhook",
    "targetUrl": "https://my-machine.example.com",
  },
  Object {
    "canDelete": false,
    "canEdit": false,
    "enabled": false,
    "id": 2,
    "name": "your-webhook",
    "targetUrl": "https://your-machine.example.com",
  },
]
`;
